modules/sdk-coin-hbar/src/hbar.ts

```diff
@@ -249,8 +286,134 @@ export class Hbar extends BaseCoin {
+        );
+      }
+    }
+
+    // validate userKey for unsigned sweep
+    if (isUnsignedSweep && !Utils.isValidPublicKey(params.userKey)) {
+      throw new Error('invalid userKey, got: ' + params.userKey);
+    }
+
+    // validate backupKey for unsigned sweep
+    if (isUnsignedSweep && !Utils.isValidPublicKey(params.backupKey)) {
+      throw new Error('invalid backupKey, got: ' + params.backupKey);
+    }
+
+    const { address: destinationAddress, memoId } = Utils.getAddressDetails(params.recoveryDestination);
+
+    const client = this.getHbarClient();
+
+    const balance = await this.getAccountBalance(params.rootAddress, client);
+    const nativeBalance = balance.hbars;
+
+    const fee = params.maxFee ? params.maxFee : '10000000';
+
+    if (new BigNumber(nativeBalance).isZero() || new BigNumber(nativeBalance).isLessThanOrEqualTo(fee)) {
+      throw new Error('Insufficient balance to recover, got balance: ' + nativeBalance + ' fee: ' + fee);
+    }
+
+    const nodeId = params.nodeId ? params.nodeId : '0.0.3';
+
+    const spendableAmount = new BigNumber(nativeBalance).minus(fee).toString();
+
+    const txBuilder = this.getBuilderFactory().getTransferBuilder();
+    txBuilder.node({ nodeId });
+    txBuilder.fee({ fee });
+    txBuilder.source({ address: params.rootAddress });
+    txBuilder.send({ address: destinationAddress, amount: spendableAmount });
+    txBuilder.validDuration(180);
+
+    if (memoId) {
+      txBuilder.memo(memoId);
+    }
+
+    if (isUnsignedSweep) {
+      // validate startTime
+      if (!params.startTime) {
+        throw new Error('start time is required for unsigned sweep');
+      }
+      if (!Utils.isValidTimeString(params.startTime)) {
+        throw new Error('invalid startTime, got: ' + params.startTime);
+      }
+      if (new BigNumber(params.startTime).isLessThanOrEqualTo(Date.now())) {
+        throw new Error('startTime must be a future timestamp, got: ' + params.startTime);
+      }
+
+      if (params.startTime) {
+        txBuilder.startTime(params.startTime);
+      }
+      const tx = await txBuilder.build();
+      const txJson = tx.toJson();
+      return {
+        txHex: tx.toBroadcastFormat(),
+        coin: this.getChain(),
+        id: txJson.id,
+        startTime: txJson.startTime,
+        validDuration: txJson.validDuration,
+        node: txJson.node,
+        memo: txJson.memo,
+        userKey: params.userKey,
+        backupKey: params.backupKey,
+        bitgoKey: params.bitgoKey,
+        maxFee: fee,
+        address: params.rootAddress,
+        recipients: txJson.instructionsData.params.recipients,
+        amount: txJson.amount,
+        json: txJson,
+      };
+    }
+
+    txBuilder.sign({ key: userPrv });
+    txBuilder.sign({ key: backUp });
+
+    const tx = await txBuilder.build();
+
+    const result = await this.broadcastTransaction(client, tx.toBroadcastFormat());
```

Comment:
Hbar is very sensitive to the selected nodeId and the broadcast time window, so instead of just returning the broadcasteable tx here, we broadcast the tx.

Reply:
Discussed in focus hour, we will split this out to a separate step and build a new re-usable form in WRW for this.